dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -927,214 +938,361 @@ private String getContentWDL(File tempMainDescriptor, Type type) {
     }
 
     /**
-     * This method will get the content for tool tab with descriptor type = CWL
+     * This method will get the content for tool tab or DAG tab with descriptor type = CWL
      * It will then call another method to transform the content into JSON string and return
+     * TODO: Currently only works for CWL 1.0, but should support at least draft 3 too.
      * @param content has the content of main descriptor file
      * @param secondaryDescContent has the secondary files and the content
      * @param type either dag or tools
      * @return String
      * */
+    @SuppressWarnings("checkstyle:methodlength")
```

Comment:
It probably makes sense to do some clean-up to see if this can be broken up into re-usable methods rather than suppressing the warning.